src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -343,6 +346,30 @@ request_creator::make_delete_objects_request(
         std::make_unique<delete_objects_body>(std::move(body))}}};
 }
 
+std::string
+request_creator::make_host([[maybe_unused]] const bucket_name& name) const {
+    switch (_ap_style) {
+    case s3_url_style::virtual_host:
+        // Host: bucket-name.s3.region-code.amazonaws.com
+        return fmt::format("{}.{}", name(), _ap());
+    case s3_url_style::path:
+        // Host: s3.region-code.amazonaws.com
+        return fmt::format("{}", _ap());
```

Comment:
why the bucket name is not included?

Reply:
Following the [S3 User Guide's examples of path-style requests](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#VirtualHostingExamples), this seems to be the proper form for the host in the header.

The bucket name is instead used in the target of the header.

In commit 3d4ef12e2c4b7d69ddec9a838e9114bf8372be47, I added some comments in the various `request_creator::make_*()` functions to reflect this style.

Reply:
> Following the [S3 User Guide's examples of path-style requests](https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#VirtualHostingExamples), this seems to be the proper form for the host in the header.
> 
> The bucket name is instead used in the target of the header.
> 
> In commit [8edf7d5](https://github.com/redpanda-data/redpanda/commit/8edf7d571c2050cdf212135a65020f52d055655e), I added some comments in the various `request_creator::make_*()` functions to reflect this style.

I was also confused here. But the answer is that this is `make_host`, not `make_target`, right?

Reply:
Yes, correct. This is `make_host`.